lib/iris/fileformats/dot.py

```diff
@@ -33,20 +33,35 @@
 _GRAPH_INDENT = ' ' * 4
 _SUBGRAPH_INDENT = ' ' * 8
 
-_DOT_EXECUTABLE_PATH = iris.config.get_option('System', 'dot_path',
-                                              default='dot')
-if not os.path.exists(_DOT_EXECUTABLE_PATH):
-    if not os.path.isabs(_DOT_EXECUTABLE_PATH):
-        try:
-            # Check PATH
-            subprocess.check_output([_DOT_EXECUTABLE_PATH, '-V'],
-                                    stderr=subprocess.STDOUT)
-        except OSError:
-            _DOT_EXECUTABLE_PATH = None
+
+_DOT_CHECKED = False
+_DOT_EXECUTABLE_PATH = None
+
+
+def _dot_path():
+    global _DOT_CHECKED, _DOT_EXECUTABLE_PATH
+
+    if _DOT_CHECKED:
+        path = _DOT_EXECUTABLE_PATH
     else:
-        _DOT_EXECUTABLE_PATH = None
+        path = iris.config.get_option('System', 'dot_path', default='dot')
+        if not os.path.exists(path):
+            if not os.path.isabs(path):
+                try:
+                    # Check PATH
+                    subprocess.check_output([path, '-V'],
+                                            stderr=subprocess.STDOUT)
+                except (OSError, subprocess.CalledProcessError):
```

Comment:
I met the same problem when I import `iris`. I edited the `dot.py` taking @rhattersley's work  as reference. But what beneath the ` except (OSError, subprocess.CalledProcessError):`?   


    ` 36 _DOT_EXECUTABLE_PATH = iris.config.get_option('System', 'dot_path',
     37                                               default='dot')
     38 if not os.path.exists(_DOT_EXECUTABLE_PATH):
     39     if not os.path.isabs(_DOT_EXECUTABLE_PATH):
     40         try:
     41             # Check PATH
     42             subprocess.check_output([_DOT_EXECUTABLE_PATH, '-V'],
     43                                     stderr=subprocess.STDOUT)
     44         except OSError:
     45             _DOT_EXECUTABLE_PATH = None
     46     else:
     47         path = iris.config.get_option('System', 'dot_path', default='dot')
     48         if not os.path.exists(path):
     49             if not os.path.isabs(path):
     50                 try:
     51                     # Check PATH
     52                     subprocess.check_output([path, '-V'],
     53                                             stderr=subprocess.STDOUT)
     54                 except (OSError, subprocess.CalledProcessError):
     55 # Edit in 20160606
     56 #        _DOT_EXECUTABLE_PATH = None
     57 #: Whether the 'dot' program is present (required for "dotpng" output).
     58 #DOT_AVAILABLE = _DOT_EXECUTABLE_PATH is not None
     59 DOT_AVAILABLE = _dot_path() is not None
     60 
     61 def save(cube, target):
     62     """Save a dot representation of the cube.
     63 
     64     Args:`